static/js/settings_account.js

```diff
@@ -63,25 +63,73 @@ function update_user_custom_profile_fields(fields, method) {
                                    {data: JSON.stringify(fields)}, spinner);
 }
 
-exports.add_custom_profile_fields_to_settings = function () {
-    if (!overlays.settings_open()) {
-        return;
-    }
+exports.initialize_custom_date_type_field = function (element) {
```

Comment:
I'd pluralize this, and also split the refactor into a bunch of small commits, where each one just extracts one function or rearranges one block of code; it's a lot easier to review that way (and I would also argue, a lot harder to make mistakes doing it).